contracts/standards/CurioTreaty.sol

```diff
@@ -32,32 +18,59 @@ abstract contract CurioTreaty is ITreaty {
     }
 
     modifier onlyOwner() {
+        GetterFacet getter = GetterFacet(diamond);
         uint256 treatyID = getter.getEntityByAddress(address(this));
         uint256 ownerID = abi.decode(getter.getComponent("Owner").getBytesValue(treatyID), (uint256));
         require(ownerID == getter.getEntityByAddress(msg.sender), "CurioTreaty: Only owner can call");
         _;
     }
 
     modifier onlySigner() {
+        GetterFacet getter = GetterFacet(diamond);
         require(getter.getNationTreatySignature(getter.getEntityByAddress(msg.sender), getter.getEntityByAddress(address(this))) != 0, "CurioTreaty: Only signer can call");
         _;
     }
 
     /// @dev No need to use if the treaty does not need whitelist
     modifier onlyWhitelist() {
+        GetterFacet getter = GetterFacet(diamond);
         require(getter.isWhitelistedByTreaty(getter.getEntityByAddress(msg.sender), getter.getEntityByAddress(address(this))), "CurioTreaty: Only whitelisted nations can call");
         _;
     }
 
+    /**
+     * @dev Check if a nation has been a treaty signer for at least a duration.
+     * @param _duration duration in seconds
+     */
+    function minimumStay(uint256 _duration) public view returns (bool) {
+        GetterFacet getter = GetterFacet(diamond);
+        uint256 nationID = getter.getEntityByAddress(msg.sender);
+        uint256 treatyID = getter.getEntityByAddress(address(this));
+
+        // Check if nation has been a treaty signer for at least the duration
+        uint256 nationJoinTime = abi.decode(getter.getComponent("InitTimestamp").getBytesValue(getter.getNationTreatySignature(nationId, treatyID)), (uint256));
+        require(block.timestamp >= nationJoinTime + _duration, "Have not stayed for minimum duration");
+        _;
+    }
+
     // ----------------------------------------------------------
-    // MEMBERSHIP FUNCTIONS (CALLED BY NATION)
+    // TREATY INFO (CALLED BY NATION)
     // ----------------------------------------------------------
 
+    function name() external view virtual returns (string memory);
+    function description() external view virtual returns (string memory);
+
+    // ----------------------------------------------------------
+    // MEMBERSHIP FUNCTIONS (CALLED BY NATION)
+    // ----------------------------------------------------------
+    
     /**
      * @dev Join treaty. Must be called by nation.
      */
     function treatyJoin() public virtual {
         // Add signature
+        GetterFacet getter = GetterFacet(diamond);
+        AdminFacet admin = AdminFacet(diamond);
```

Comment:
You'd need to import `AdminFacet` at the beginning of the contract. In the future, you can run `yarn ft` before committing to double check logic, especially things like `CurioTreaty` which already has many tests.